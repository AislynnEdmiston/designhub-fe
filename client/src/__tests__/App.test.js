const puppeteer = require('puppeteer');
jest.setTimeout(30000);

test('renders without crashing', () => {
  const condition = 1 + 2;
  expect(condition).toEqual(3);
});

test('login with auth0', async () => {
  const browser = await puppeteer.launch({
    headless: false
  });
  const page = await browser.newPage();
  await page.goto('http://localhost:3000');
  await page.click('.auth0-redirect-btn');

  await page.waitForSelector('input[name="email"]', {
    visible: true,
    timeout: 5000
  });
  await page.type('input[name="email"]', 'test@test.com', { delay: 50 });
  await page.type('input[name="password"]', 'Test1234', { delay: 50 });

  await page.click('button[type="submit"]');
  await page.waitFor('.edit-profile-btn');
  const text = await page.$eval('.edit-profile-btn', el => el.innerHTML);
  expect(text).toEqual('Edit Profile');
});
