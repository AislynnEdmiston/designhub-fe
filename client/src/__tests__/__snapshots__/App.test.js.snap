// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> should match the previous snapshot of the component tests that the component has not changed and matched the previous snapshot 1`] = `
Array [
  <div
    className="Login-container"
  >
    <section
      className="left-side"
    >
      <p>
        WELCOME TO
      </p>
      <h1>
        DESIGN
        <em>
          HUB
        </em>
      </h1>
      <div
        className="paragraphs"
      >
        <span
          aria-label="handshake"
          role="img"
        >
          ü§ù
        </span>
        <h2>
          Join and start creating projects and collaborating with teams!
        </h2>
        <span
          aria-label="briefcase"
          role="img"
        >
          üíº
        </span>
        <h2>
          House your case studies, deliverables, assets, links, and design files all in one place!
        </h2>
        <span
          aria-label="megaphone"
          className="megaphone"
          role="img"
        >
          üì¢
        </span>
        <h2>
          Give and recieve feedback with ease. No more worrying about your feedback getting lost in Slack messages!
        </h2>
      </div>
    </section>
    <section
      className="right-side"
    >
      <div
        className="cta"
      >
        <img
          alt="logo"
          className="logo"
          src="SampleLogo.svg"
        />
        <h1>
          Super-charge your design workflow now and start collaborating like a beast
        </h1>
        <div>
          <h6
            className="join-head"
          >
            Join DesignHub today.
          </h6>
          <button
            className="auth0-redirect-btn"
            onClick={[Function]}
          >
            Create an account or Sign in
          </button>
        </div>
      </div>
    </section>
  </div>,
  <div
    className="App"
  />,
]
`;
